crates/viewer/re_viewer/src/ui/settings_screen.rs

```diff
@@ -160,6 +138,102 @@ fn settings_screen_ui_impl(ui: &mut egui::Ui, app_options: &mut AppOptions, keep
+
+                ui.label("Path:");
+
+                ui.add(egui::TextEdit::singleline(
+                    &mut app_options.video_decoder_ffmpeg_path,
+                ));
+            });
+        });
+
+        ffmpeg_path_status_ui(ui, app_options);
+    }
+
+    // This affects only the web target, so we don't need to show it on native.
+    #[cfg(target_arch = "wasm32")]
+    {
+        let hardware_acceleration = &mut app_options.video_decoder_hw_acceleration;
+        ui.horizontal(|ui| {
+            ui.label("Decoder:");
+            egui::ComboBox::from_id_salt("video_decoder_hw_acceleration")
+                .selected_text(hardware_acceleration.to_string())
+                .show_ui(ui, |ui| {
+                    ui.selectable_value(
+                        hardware_acceleration,
+                        re_video::decode::DecodeHardwareAcceleration::Auto,
+                        re_video::decode::DecodeHardwareAcceleration::Auto.to_string(),
+                    ) | ui.selectable_value(
+                        hardware_acceleration,
+                        re_video::decode::DecodeHardwareAcceleration::PreferSoftware,
+                        re_video::decode::DecodeHardwareAcceleration::PreferSoftware.to_string(),
+                    ) | ui.selectable_value(
+                        hardware_acceleration,
+                        re_video::decode::DecodeHardwareAcceleration::PreferHardware,
+                        re_video::decode::DecodeHardwareAcceleration::PreferHardware.to_string(),
+                    )
+                });
+            // Note that the setting is part of the video's cache key, so, if it changes, the cache
+            // entries outdate automatically.
+        });
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn ffmpeg_path_status_ui(ui: &mut Ui, app_options: &AppOptions) {
+    use re_video::decode::{FFmpegVersion, FFmpegVersionParseError};
+
+    let path = app_options
+        .video_decoder_override_ffmpeg_path
+        .then(|| std::path::Path::new(&app_options.video_decoder_ffmpeg_path));
+
+    if let Some(path) = path {
+        if !path.is_file() {
+            ui.error_label("The specified FFmpeg binary path does not exist or is not a file.");
+            return;
+        }
+    }
+
+    let res = FFmpegVersion::for_executable(path);
+    match res {
+        Ok(version) => {
+            if version.is_compatible() {
+                ui.success_label(&format!("FFmpeg found (version {version})"));
+            } else {
+                ui.error_label(&format!("Incompatible FFmpeg version: {version}"));
+            }
+        }
+        Err(FFmpegVersionParseError::ParseVersion { raw_version }) => {
+            ui.warning_label(&format!(
+                "FFmpeg binary found but unable to parse version: {raw_version}"
+            ));
+        }
+
+        Err(err) => {
+            ui.error_label(&format!("Unable to check FFmpeg version: {err}"));
```

Comment:
Why are some of these errors and some warnings? 🤔

Reply:
This one is a warning because we do let it slip in the code, as version checking is flaky and it might still end up working. I'll add a clarifying comment.